Kitodo/src/main/java/org/kitodo/model/Subfolder.java

```diff
@@ -0,0 +1,301 @@
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.bind.JAXBException;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.kitodo.api.filemanagement.FileManagementInterface;
+import org.kitodo.config.ConfigCore;
+import org.kitodo.config.xml.fileformats.FileFormat;
+import org.kitodo.config.xml.fileformats.FileFormatsConfig;
+import org.kitodo.data.database.beans.Folder;
+import org.kitodo.data.database.beans.Process;
+import org.kitodo.helper.VariableReplacer;
+import org.kitodo.metadata.comparator.MetadataImageComparator;
+import org.kitodo.serviceloader.KitodoServiceLoader;
+
+/**
+ * A subfolder is a folder in the file system below the process folder that
+ * contains a variant of each file of the digital representation of the work
+ * intended for a particular use. In this approach, a file is a fragment of the
+ * digital representation of the artwork, a conceptual entity that can have
+ * different variants. The individual variants of this file are then each stored
+ * as computer-readable files in a folder for that use on the volume. They can
+ * have different formats, both in terms of the data type and their form of
+ * representation. The computer files, which are located in different use
+ * folders and which are conceptually the same file, share a part of their
+ * filename. This is referred to here as the canonical part of the file name.
+ * 
+ * <p>
+ * For historical reasons, in some institutions using this software, there is a
+ * situation where the different variants of a file are in the same folder and
+ * differ in filename suffix. We do not want to advertise this setup, but we
+ * will continue to support it because it would be difficult for these
+ * institutions to migrate the files.
+ */
+public class Subfolder {
+    /**
+     * The general metrics of this kind of subfolder. So to say its type.
+     */
+    private final Folder folder;
+
+    /**
+     * The process this subfolder belongs to. That is the process whose process
+     * directory this sub-folder is below on the file storage.
+     */
+    private final Process process;
+
+    /**
+     * The variable replacer used to replace variables in the folder path to the
+     * task folder.
+     */
+    private final VariableReplacer variableReplacer;
+
+    /**
+     * Convenience function to create a bunch of subfolders in one.
+     * 
+     * @param process
+     *            the process this subfolder belongs to
+     * @param folders
+     *            The general metrics of the kinds of subfolders to create
+     * @return subfolders
+     */
+    public static List<Subfolder> createAll(Process process, Collection<Folder> folders) {
```

Comment:
Please don't create a bunch of entitys from the class itself. Should be moved to a service.